File: src/main/java/dijkstra/application/SquareGraph.java
package dijkstra.application;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.function.Supplier;
import org.jgrapht.Graph;
import org.jgrapht.GraphPath;
import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
import org.jgrapht.generate.GridGraphGenerator;
import org.jgrapht.graph.SimpleWeightedGraph;
import org.jgrapht.traverse.DepthFirstIterator;
import org.jgrapht.util.SupplierUtil;

public final class SquareGraph {
    private final Supplier vertexSupplier;
    private final Graph graph;
    private int rows;
    private int columns;

    private final Graph initialize() {
        SimpleWeightedGraph completeGraph = new SimpleWeightedGraph(this.vertexSupplier, SupplierUtil.createDefaultWeightedEdgeSupplier());
        GridGraphGenerator graphGenerator = new GridGraphGenerator(this.rows, this.columns);
        graphGenerator.generateGraph((Graph)completeGraph);
        return (Graph)completeGraph;
    }

    public final GraphPath dijkstra(String start, String end) {
        DijkstraShortestPath dijkstra = new DijkstraShortestPath(this.graph);
        return dijkstra.getPath(start, end);
    }

    public final void deleteVerteces( ArrayList vertices) {
        Iterator var3 = vertices.iterator();

        while(var3.hasNext()) {
            String v = (String)var3.next();
            this.graph.removeVertex(v);
        }

    }


    public String toString() {
        String str = "";

        String vertex;
        for(Iterator iter = (Iterator)(new DepthFirstIterator(this.graph));
            iter.hasNext(); str = str + "Vertice `" + vertex +
                "` connesso a " + this.graph.edgesOf(vertex).toString() + "\n")
        {
            vertex = (String)iter.next();
        }

        return str;
    }

    public final int getRows() {
        return this.rows;
    }

    public final void setRows(int var1) {
        this.rows = var1;
    }

    public final int getColumns() {
        return this.columns;
    }

    public final void setColumns(int var1) {
        this.columns = var1;
    }

    public SquareGraph(int rows, int columns) {
        this.rows = rows;
        this.columns = columns;
        this.vertexSupplier = (Supplier) (new Supplier() {
            private int id;


            public String get() {
                StringBuilder var10000 = (new StringBuilder()).append("Vertex ");
                int var1;
                this.id = (var1 = this.id) + 1;
                return var10000.append(var1).toString();
            }
        });
        this.graph = this.initialize();
        if (this.rows < 2) {
            this.rows = 2;
        }

        if (this.columns < 2) {
            this.columns = 2;
        }

    }
}

File: src/main/kotlin/dijkstra/SquareGraph.kt
package dijkstra

import org.jgrapht.Graph
import org.jgrapht.GraphPath
import org.jgrapht.alg.shortestpath.DijkstraShortestPath
import org.jgrapht.generate.GridGraphGenerator
import org.jgrapht.graph.DefaultWeightedEdge
import org.jgrapht.graph.SimpleWeightedGraph
import org.jgrapht.traverse.DepthFirstIterator
import org.jgrapht.util.SupplierUtil
import java.util.function.Supplier

class SquareGraph(var rows: Int = 2, var columns: Int = 2){

    private val vertexSupplier: Supplier<String> = object : Supplier<String> {
        private var id = 0
        override fun get(): String {
            return "Vertex " + id++;
        }
    }

    private val graph = initialize();

    init{
        if(rows < 2) rows = 2;
        if(columns < 2) columns = 2;
    }

    private fun initialize() : Graph<String, DefaultWeightedEdge> {
        var completeGraph =
            SimpleWeightedGraph(vertexSupplier, SupplierUtil.createDefaultWeightedEdgeSupplier());

        // Successivamente si crea un generatore che crea la struttura di `completeGraph`
        var graphGenerator =
            GridGraphGenerator<String, DefaultWeightedEdge>(rows, columns);

        graphGenerator!!.generateGraph(completeGraph)
        return completeGraph;
    }


    fun dijkstra(start : String, end : String): GraphPath<String, DefaultWeightedEdge>? {
        var dijkstra = DijkstraShortestPath(graph);
        return dijkstra.getPath(start, end);
        ;
    }

    fun deleteVerteces(vertices : ArrayList<String>){
        for(v in vertices){
            graph.removeVertex(v);
        }
    }

    override fun toString(): String {
        var str = ""
        val iter: Iterator<String> = DepthFirstIterator(graph)
        while (iter.hasNext()) {
            val vertex = iter.next()
            str += "Vertice `" + vertex + "` connesso a " + graph.edgesOf(vertex).toString() + "\n";
        }
        return str
    }
}